File: backend-python/app/models/product_sku.py
from datetime import datetime
from decimal import Decimal

from sqlalchemy import BigInteger, DateTime, Integer, String, text
from sqlalchemy.dialects.mysql import DECIMAL
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class ProductSku(Base):
    __tablename__ = "product_sku"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    product_id: Mapped[int] = mapped_column(BigInteger, nullable=False)
    external_sku_id: Mapped[str] = mapped_column(String(64), nullable=False)
    sku_name: Mapped[str | None] = mapped_column(String(255), nullable=True)
    sku_attr: Mapped[str | None] = mapped_column(String(255), nullable=True)
    sale_price: Mapped[Decimal] = mapped_column(DECIMAL(10, 2), nullable=False)
    cost_price: Mapped[Decimal] = mapped_column(DECIMAL(10, 2), nullable=False)
    stock: Mapped[int] = mapped_column(Integer, nullable=False, default=0, server_default="0")
    created_at: Mapped[datetime] = mapped_column(DateTime, nullable=False, server_default=text("CURRENT_TIMESTAMP"))
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, server_default=text("CURRENT_TIMESTAMP"), server_onupdate=text("CURRENT_TIMESTAMP")
    )

File: backend-java/src/main/java/com/example/pricing/entity/ProductSku.java
package com.example.pricing.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 商品 SKU 实体，记录规格级别的价格、属性和库存信息。
 */
@Data
@TableName("product_sku")
public class ProductSku {
    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("product_id")
    private Long productId;

    @TableField("external_sku_id")
    private String externalSkuId;

    @TableField("sku_name")
    private String skuName;

    @TableField("sku_attr")
    private String skuAttr;

    @TableField("sale_price")
    private BigDecimal salePrice;

    @TableField("cost_price")
    private BigDecimal costPrice;

    @TableField("stock")
    private Integer stock;

    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
